modules/javafx.graphics/src/main/java/javafx/scene/robot/Robot.java

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright (c) 2010, 2017, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package javafx.scene.robot;
+
+import static com.sun.javafx.FXPermissions.CREATE_ROBOT_PERMISSION;
+
+import java.util.Objects;
+
+import javafx.geometry.Point2D;
+import javafx.geometry.Rectangle2D;
+import javafx.scene.image.Image;
+import javafx.scene.image.WritableImage;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.MouseButton;
+import javafx.scene.paint.Color;
+
+import com.sun.glass.ui.GlassRobot;
+import com.sun.javafx.tk.Toolkit;
+
+/**
+ * A {@code Robot} is used for simulating user interaction such as
+ * typing keys on the keyboard and using the mouse as well as capturing
+ * graphical information without requiring a {@link javafx.scene.Scene}
+ * instance.
+ *
```

Comment:
This would be a good place to add a comment about threading requirements. You can use something like the following, taken from Stage:

Robot objects must be constructed and modified on the JavaFX Application Thread.

Reply:
Although "modified" might be better changed to "used" since Robot is stateless.